### PR TITLE
Corrected autocomplete cursor position in data filter tab

### LINKED ISSUE #22315: Autocomplete misplaces the cursor
### Description

Hi all,

Auto-completion of column names incorrectly moves the cursor if there is other text on the line.

Observe the position of the cursor | in the following screencast

dbeaver

I want a where clause like this:
account_status in ('ACTIVE',  'SUSPENDED')
I get this:
account_status ('ACinTIVE',  'SUSPENDED')

### DBeaver Version

Community Edition 23.3.1

### Operating System

Windows 10

### Database and driver

_No response_

### Steps to reproduce

1. Click on 'Filter panel'
2. Write or paste the text ('ACTIVE', 'SUSPENDED')
3. Move the cursor at the beginning of the line and start writing the first **3 or 4** characters of the column name (the menu with the possible columns name appears)
4. Using the arrow keys move to the column name and press enter, the column name is entered and the cursor moves by **3 or 4** **(it appears to move by the number of characters used in step 3)** characters beyond the name 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sarveshhiwase**
### Resolved issue #22315 of the misplaced cursor while autocompletion in the data filter tab.

Explanation - 
Database with 1 table with 3 columns - (country_id, country_name, region_id)
When the user hits enter to accept the proposal for autocomplete, it replaces the cou and completes it with country_id, 
but now the cursor position is not at the end of country_id but 3 more characters from country_id, why is that, because the time 
at which the user has hit entered, the cursor position is set to the cursor position when enter was pressed + the autocomplete word length
```txt
Example 
cou is typed
result -
country_id  |
```

I feel this behavior is more semantically meaningful because when we type in IDEs, we find that after pressing **`SPACE`** the autocomplete mostly stops and starts afresh for new words after the space.

### UI Changes - 
- ### Before - 
  - screenshot-1
  - #22315-before*Please open in new tab, for better quality, sorry for bad quality*
- ### After -
  - #22315-after*Please open in new tab, for better quality, sorry for bad quality*

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetFilterContentAdapter.java [hunk 1]
```diff
@@ -61,11 +63,9 @@ public void insertControlContents(Control control, String contents, int cursorPo
         }
         text.insert(contents);
 
-        // Insert will leave the cursor at the end of the inserted text. If this
-        // is not what we wanted, reset the selection.
-        if (cursorPosition <= contents.length()) {
-            text.setSelection(selection.x + cursorPosition,
-                selection.x + cursorPosition);
+        // Always insert the cursor at the end of inserted text
+        if(wordStartingPosition != -1) {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetFilterContentAdapter.java]:
```suggestion
        if (wordStartingPosition != -1) {
```

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetFilterContentAdapter.java [hunk 2]
```diff
@@ -61,11 +63,9 @@ public void insertControlContents(Control control, String contents, int cursorPo
...
+            text.setSelection(wordStartingPosition + contents.length(),wordStartingPosition + contents.length());
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetFilterContentAdapter.java]:
```suggestion
            text.setSelection(wordStartingPosition + contents.length(), wordStartingPosition + contents.length());
```

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetFilterPanel.java [hunk 3]
```diff
@@ -238,6 +238,9 @@ public void keyPressed(KeyEvent e) {
                         }
                         e.doit = false;
                         setCustomDataFilter();
+                    } else if(e.keyCode == SWT.SPACE) {
+                        // close the proposal window, if space is pressed.
+                        filtersProposalAdapter.closeProposalPopup();
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetFilterPanel.java]:
```suggestion
                    } else if (e.keyCode == SWT.SPACE) {
                        // close the proposal window, if space is pressed.
                        filtersProposalAdapter.closeProposalPopup();
```

**Review by E1izabeth — COMMENTED** [review]
Please fix the code style

**Comment by sarveshhiwase** [comment]
@E1izabeth Thanks for the review, I'll make the changes and commit.

Can you also let me know from where I can know the code style, so I won't miss them next time?

Edited -
Updated with proper code style.

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by E1izabeth** [comment]
Pull Request passed to QA team for testing

**Comment by uslss** [comment]
verified

**Pull request merged**